block-layer/codecs/DAG-JSON.md

```diff
@@ -14,14 +14,56 @@ Contrary to popular belief, JSON as a format supports Big Integers. It's only
 JavaScript itself that has trouble with them. This means JS implementations
 of `dag-json` can't use the native JSON parser and serializer.
 
+### Version 0
+
+This is an old version of `dag-json` that reserved the `"/"` key in order to
+represent binary and link data types.
+
 #### Binary Type
 
 ```javascript
 {"/": { "base64": String }}
 ```
 
-### Link Type
+#### Link Type
 
 ```javascript
 {"/": String /* base encoded CID */}
-```
\ No newline at end of file
+```
+
+### Version 1
+
+#### Format
+
+The internal data format is valid JSON but is **NOT** identical to the decoded
+node data codecs produce.
+
+Example internal format:
+
+```javascript
+{ "/": { "_": 1 },
+  "data": { "hello": "world", { "obj": { "array": [0, 0] } } },
+  "meta": {
+    base64: [
+      [[ "key" ], "dmFsdWU="],
+      [[ "obj", "key"], "dmFsdWU="],
+      [[ "obj", "array", 0], "dmFsdWU="]
+    ],
+    links: [
+      [["obj", "array", 1], "zb2rhe5P4gXftAwvA4eXQ5HJwsER2owDyS9sKaQRRVQPn93bA"]
+    ]
+  }
+}
+```
+
+Decodes to:
+
+```javascript
+{ hello: 'world',
+  key: Buffer.from('value'),
```

Comment:
I would really prefer if we'd move to an Browser first approach and not using Node.js primitives:

```JS
    key: new TextEncoder().encode('value'),
```

Reply:
it’s just not entirely clear to everyone that this does binary encoding. perhaps we should just move to psuedo-code, something like `Bytes(‘value’)`

Reply:
That would work for me. I just don't want to give the impression it's a Node.js Buffer.